src/error.rs

```diff
@@ -49,7 +49,72 @@ pub struct Pos {
     path: String,
 }
 
+
+/// This type represents the location that an error occured
+#[derive(Debug)]
+pub struct ErrorMarker {
+    index: usize,
+    line: usize,
+    col: usize,
+} 
+
+impl ErrorMarker {
+    /// The byte index of the error
+    pub fn index(&self) -> usize {
+        self.index
+    }
+
+    /// The line of the error
+    pub fn line(&self) -> usize {
+        self.line
+    }
+
+    /// The column of the error
+    pub fn col(&self) -> usize {
+        self.col
+    }
+
+    fn from_marker(marker: &Marker) -> Self {
```

Comment:
Can you add a comment that this is required to keep Marker private?

Alternatively, you can impl From and add `#[doc(hidden)]`.